rtos/Mutex.h

```diff
@@ -73,13 +75,28 @@ class Mutex : private mbed::NonCopyable<Mutex> {
     /** Create and Initialize a Mutex object
 
      @param name name to be used for this mutex. It has to stay allocated for the lifetime of the thread.
-
      @note You cannot call this function from ISR context.
     */
     Mutex(const char *name);
 
-    /** Wait until a Mutex becomes available.
-      @param   millisec  timeout value or 0 in case of no time-out. (default: osWaitForever)
+    /**
+      Wait until a Mutex becomes available.
+
+      @return  status code that indicates the execution status of the function:
+               @a osOK the mutex has been obtained.
+
+      @note You cannot call this function from ISR context.
+      @note This function asserts status of the lock operation (will not return in case of failure).
```

Comment:
Avoid the word "assert", as it's no longer an assert in the C sense. Maybe  "This function treats RTOS errors as fatal system errors, so can only return osOK".

Reply:
Fixed as suggested.